frame/staking-rewards/src/lib.rs

```diff
@@ -780,9 +773,98 @@ pub mod pallet {
 			});
 			Ok([*position, new_position])
 		}
+
+		#[transactional]
+		fn claim(who: &Self::AccountId, position: &Self::PositionId) -> DispatchResult {
+			let keep_alive = false;
+
+			let mut stake =
+				Stakes::<T>::try_get(position).map_err(|_| Error::<T>::StakeNotFound)?;
+			ensure!(who == &stake.owner, Error::<T>::OnlyStakeOwnerCanUnstake);
+
+			let pool_id = stake.reward_pool_id;
+			let mut rewards_pool =
+				RewardPools::<T>::try_get(pool_id).map_err(|_| Error::<T>::RewardsPoolNotFound)?;
+
+			(rewards_pool, stake) =
+				Self::collect_rewards(&pool_id, rewards_pool, stake, false, keep_alive)?;
+
+			RewardPools::<T>::insert(pool_id, rewards_pool);
+			Stakes::<T>::insert(position, stake);
+
+			Self::deposit_event(Event::<T>::Claimed { owner: who.clone(), position_id: *position });
+
+			Ok(())
+		}
 	}
 
 	impl<T: Config> Pallet<T> {
+		/// Transfers the rewards a staker has earned while updating the provided `rewards_pool`.
+		///
+		/// # Params
+		/// * `pool_id` - Pool identifier
+		/// * `mut rewards_pool` - Rewards pool to update
+		/// * `stake` - Stake position
+		/// * `early_unlock` - If there should be an early unlock penalty
+		/// * `keep_alive` - If the transaction should be kept alive
+		pub(crate) fn collect_rewards(
+			pool_id: &T::RewardPoolId,
+			mut rewards_pool: RewardPoolOf<T>,
+			mut stake: StakeOf<T>,
+			early_unlock: bool,
+			keep_alive: bool,
+		) -> Result<(RewardPoolOf<T>, StakeOf<T>), DispatchError> {
+			let mut inner_rewards = rewards_pool.rewards.into_inner();
+
+			for (asset_id, reward) in inner_rewards.iter_mut() {
```

Comment:
it's possible to directly iter_mut over the bounded map now

Reply:
+1
7c9c683e2b26a367d45c9fe9d43226dbeb0b37cb